tests/unit/s2n_ecc_x25519_test.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include <s2n.h>
+
+#include "crypto/s2n_ecc_x25519.h"
+#include "utils/s2n_mem.h"
+#include "testlib/s2n_testlib.h"
+#include "stuffer/s2n_stuffer.h"
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+    {
+        /* Test compute shared sceret for Curve25519 */
+        struct s2n_ecc_evp_params ecc_evp_params;
+        struct s2n_blob wire, server_shared, client_shared;
+
+        ecc_evp_params.negotiated_curve = &s2n_X25519;
+
+        /* Server generates a key */
+        EXPECT_SUCCESS(s2n_ecc_evp_generate_ephemeral_key(&ecc_evp_params));
+
+        /* Client generates a key, computes its shared secret and sends the client public key */
+        EXPECT_SUCCESS(s2n_ecc_evp_compute_shared_secret_as_client(&ecc_evp_params, &wire, &client_shared));
+
+        /* Server receives the client public key and computes its shared secret */
+        EXPECT_SUCCESS(s2n_ecc_evp_compute_shared_secret_as_server(&ecc_evp_params, &wire, &server_shared));
+
```

Comment:
have you considered testing the failure/negative cases?

Reply:
Good suggestion, will add.